### PR TITLE
deps(fix): update to 2.16.0 log4j version and ban all 2.x versions which are <= 2.15.0

### PR DESCRIPTION
**Pull request by mutianf**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigtable-hbase/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

### CODE REVIEW

**Review by igorbernstein2 — APPROVED** [review]
lgtm

File: pom.xml [hunk 1]
```diff
@@ -266,6 +266,28 @@ limitations under the License.
...
+            <id>enforce-banned-deps</id>
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <configuration>
+              <rules>
+                <!-- ban all log4j 2.x deps with CVEs -->
+                <bannedDependencies>
+                  <excludes>
+                    <exclude>org.apache.logging.log4j:*:[2.0-beta9, 2.14.1]</exclude>
```
Comment (kolea2) [inline pom.xml:283]: does this also include pre beta9?

Comment (kolea2) [reply pom.xml:283]: should this be log4j-api, not log4j?

File: bigtable-dataflow-parent/bigtable-beam-import/pom.xml [hunk 2]
```diff
@@ -38,6 +39,11 @@ limitations under the License.
         <type>pom</type>
         <scope>import</scope>
       </dependency>
+      <dependency>
+        <groupId>org.apache.logging.log4j</groupId>
+        <artifactId>log4j-api</artifactId>
```
Comment (kolea2) [inline bigtable-dataflow-parent/bigtable-beam-import/pom.xml:44]: what about log4j-core?

File: pom.xml [hunk 3]
```diff
@@ -266,6 +266,28 @@ limitations under the License.
           <ignoreNonCompile>true</ignoreNonCompile>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-enforcer-plugin</artifactId>
```
Comment (kolea2) [inline pom.xml:271]: is this overriding the configuration in the parent pom?

Comment (mutianf) [reply pom.xml:271]: The checks in parent pom should still be propagated unless we specifically disable it according to https://maven.apache.org/guides/mini/guide-configuring-plugins.html#using-the-inherited-tag-in-build-plugins

I also tested by  removing https://github.com/googleapis/java-bigtable-hbase/blob/bigtable-1.x/bigtable-dataflow-parent/bigtable-beam-import/pom.xml#L259 and got the dependency errors

File: bigtable-dataflow-parent/bigtable-beam-import/pom.xml (hunk 2 above)
Comment (mutianf) [reply bigtable-dataflow-parent/bigtable-beam-import/pom.xml:44]: I think it only has log4j-api dependency here, log4j-api should be included in the banned dependencies

File: pom.xml (hunk 1 above)
Comment (mutianf) [reply pom.xml]: it's excluding `org.apache.logging.log4j:*` so should include all the artifacts?

File: pom.xml (hunk 3 above)
Comment (kolea2) [reply pom.xml:271]: excellent, thank you :)

File: pom.xml (hunk 1 above)
Comment (kolea2) [reply pom.xml]: should it be `log4j*` rather than `log4j:*`? Sorry I'm not sure how this field works, just want to make sure!

Comment (mutianf) [reply pom.xml]: I think log4j:* should work, example here: https://maven.apache.org/enforcer/enforcer-rules/bannedDependencies.html

I also tested by not updating the version to 2.15 and the build failed

File: bigtable-dataflow-parent/bigtable-hbase-beam/pom.xml [hunk 4]
```diff
@@ -27,6 +27,20 @@ limitations under the License.
...
+
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>org.apache.logging.log4j</groupId>
+        <artifactId>log4j-api</artifactId>
+        <version>${log4j2.version}</version>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-hbase-beam/pom.xml:42]: This needs to be listed as a direct dependency. DependencyManagement doesn't work for libraries

File: pom.xml [hunk 5]
```diff
@@ -63,6 +63,8 @@ limitations under the License.
     <slf4j.version>1.7.25</slf4j.version>
     <commons-logging.version>1.2</commons-logging.version>
     <jsr305.version>3.0.2</jsr305.version>
+    <log4j2.version>2.15.0</log4j2.version>
```
Comment (kolea2) [inline pom.xml:66]: you can probably clean up this same property in the tools module and reference this instead.

**Review by kolea2 — CHANGES_REQUESTED** [review]

**Review by igorbernstein2 — APPROVED** [review]

File: pom.xml (hunk 5 above)
Comment (mutianf) [reply pom.xml:66]: done!

**Review by kolea2 — APPROVED** [review]

File: pom.xml [hunk 6]
```diff
@@ -266,6 +267,28 @@ limitations under the License.
...
+            <id>enforce-banned-deps</id>
+            <goals>
+              <goal>enforce</goal>
+            </goals>
+            <configuration>
+              <rules>
+                <!-- ban all log4j 2.x deps with CVEs -->
+                <bannedDependencies>
+                  <excludes>
+                    <exclude>org.apache.logging.log4j:*:[2.0-alpha1,2.14.1]</exclude>
```
Comment (kolea2) [inline pom.xml:284]: I think we should now increase this to 2.15.0 and only allow 2.16.0 +

**Review by kolea2 — CHANGES_REQUESTED** [review]

**Review by kolea2 — APPROVED** [review]

**Pull request merged**